code/modules/mob/living/status_procs.dm

```diff
@@ -418,17 +445,19 @@
 		S = apply_status_effect(/datum/status_effect/incapacitating/sleeping, amount)
 	return S
 
-///Allows us to set a permanent sleep on a player (use with caution and remember to unset it with SetSleeping() after the effect is over)
-/mob/living/proc/PermaSleeping()
-	if(SEND_SIGNAL(src, COMSIG_LIVING_STATUS_SLEEP, -1) & COMPONENT_NO_STUN)
+/////////////////////////////////// SSD SLEEPING ////////////////////////////////////
+// Puts you to sleep with a status effect distinct from normal sleeping to avoid collision
+
+/mob/living/proc/AdjustCryoSleeping(amount) //Adds to remaining duration
+	if(SEND_SIGNAL(src, COMSIG_LIVING_STATUS_SLEEP, amount) & COMPONENT_NO_STUN)
 		return
 	if(status_flags & GODMODE)
 		return
 	var/datum/status_effect/incapacitating/sleeping/S = IsSleeping()
 	if(S)
-		S.duration = -1
-	else
-		S = apply_status_effect(/datum/status_effect/incapacitating/sleeping, -1)
+		S.duration += amount
+	else if(amount > 0)
+		S = apply_status_effect(/datum/status_effect/incapacitating/sleeping, amount)
 	return S
```

Comment:
Is there a reason `PermaSleeping` is removed? If not, I'd recommend against removing/modifying code where not necessary, to reduce conflicts when updating from upstream.

Also, I've seen a mention outside of github that the code doesn't put the player to sleep, but instead knocks them out - if that's the case, what does `AdjustCryoSleeping` do? It seems to apply regular sleep, not cryosleep.

Reply:
Honestly, I thought Singulo added it. It should probably not be removed, but it was completely unused and isn't causing any compilation errors, so...